Add NetworkConnection enumeration

nonsensery opened the pull request:
#### What does this PR do?

Adds a new enumeration type, `NetworkConnection`, to represent the type of connection that is currently available.

#### Any background context you want to provide?

This is meant to address issue #4. In that issue, I had suggested adding a "reachability" type, with values for offline, cellular and WiFi connections. Instead, I chose to model this as a "connection" type, where "offline" is represented by `nil`, and "cellular" and "WiFi" are represented by values.

By introducing a new property, I was able to maintain full backwards compatibility in the API.

Because the property is an optional value type, the `connection` property is not available to objc.

angelodipaolo commented:
Thanks for adding this! Could you add a section a to a readme about how to use the new `NetworkConnection` API?

nonsensery commented:
OK, I updated the Read Me and example app code. Since the `connection` property covers all of the functionality of `isReachable`, `isCellular` and `isWiFi`, I updated the sample code to just use `connection`.

angelodipaolo commented:
Thanks for also updating the example project
